src/main/java/com/aws/greengrass/shadowmanager/ShadowManagerDatabase.java

```diff
@@ -101,15 +102,18 @@ private void migrateDB(Flyway flyway) throws IOException, FlywayException {
 
     private void recreateDB(Flyway flyway) throws IOException {
         logger.atDebug().kv("database-path", databasePath.toString()).log("Deleting the existing DB");
-        Files.list(databasePath).forEach((path -> {
-            if (path.endsWith("db")) {
+        try (Stream<Path> workPathFiles = Files.list(databasePath)) {
+            workPathFiles.forEach(path -> {
+                if (!path.endsWith("db")) {
+                    return;
+                }
                 try {
                     Files.deleteIfExists(path);
                 } catch (IOException e) {
                     throw new RuntimeException(e);
```

Comment:
this entire method can now throw an runtime exception, I think you need to catch it and handle it in this recreateDB method.

Reply:
Why just not throw a `ShadowManagerDataException` we are re wrapping it above in it. Also why not just wrap the entire thing in a try catch, catching IOException?. This method should just throw a `ShadowManagerDataException` instead of having to add try catch all over this method and the caller code